src/main/java/pl/allegro/tech/opel/OpelEngine.java

```diff
@@ -33,9 +35,26 @@ public OpelParsingResult parse(String expression) {
 
     public CompletableFuture<?> eval(String expression) {
         ParsingResult<OpelNode> parsingResult = getParsingResult(expression);
+        if (parsingResult.hasErrors()) {
+            throw new OpelException("Error parsing expression: '" + expression + "'" + additionalErrorMsq(parsingResult));
+        }
         return parsingResult.resultValue.getValue(embeddedEvalContext);
     }
 
+    private String additionalErrorMsq(ParsingResult<OpelNode> parsingResult) {
```

Comment:
when expression has error return parsingResult.resultValue.getValue(embeddedEvalContext) throw npe. I add if and this as additional error message.